### ISSUE TITLE
Cannot pan map after using zoom navigation control

### ISSUE THREAD

**Issue by macobo** [uber/react-map-gl#700]
Writing this to save others a couple of hours.

We use react-map-gl with redux, storing viewstate in it. We ran into an issue with a pretty vanilla setup where the map stopped allowing panning after clicking on zoom navigation control.

Our component render:

```javascript
    <ReactMapGL
      onViewportChange={this.props.updateViewport}
      {...this.props.viewport}
    >
      <NavigationControl onViewportChange={this.props.updateViewport} />
```

We discovered this was because the payload passed to `this.props.updateViewport` contained a noop `onViewportChange` function.

![screenshot from 2019-01-24 18-17-43](https://user-images.githubusercontent.com/148820/51691896-62bca900-2004-11e9-990a-ab92a35fdcff.png)

This promptly got stored in our store. Due to `{...this.props.viewport}` being last this overwrote our own onViewportChange function - causing it to do nothing.

An easy fix was to move the viewport spread as the first argument:

```javascript
    <ReactMapGL
      {...this.props.viewport}
      onViewportChange={this.props.updateViewport}
    >
      <NavigationControl onViewportChange={this.props.updateViewport} />
```

Honestly no clue why this doesn't break with react setState.

**Comment by macobo** [comment]
Perfect, thank you! Feel free to close after the next release. :+1:

**Comment by Pessimistress** [comment]
Fix published in 4.0.10.
